dandi/utils.py

```diff
@@ -504,3 +510,44 @@ def delayed(*args, **kwargs):
     import joblib
 
     return joblib.delayed(*args, **kwargs)
+
+
+def get_instance(dandi_instance_id):
+    if dandi_instance_id.lower().startswith(("http://", "https://")):
+        redirector_url = dandi_instance_id
+        dandi_id = known_instances_rev.get(redirector_url)
+        if dandi_id is not None:
+            instance = known_instances[dandi_id]
+        else:
+            instance = None
+    else:
+        instance = known_instances[dandi_instance_id]
+        redirector_url = instance.redirector
+        if redirector_url is None:
+            raise ValueError("DANDI instance has no known redirector URL")
+    try:
+        r = requests.get(redirector_url.rstrip("/") + "/server-info")
+        r.raise_for_status()
+    except Exception as e:
+        lgr.warning("Request to %s failed (%s)", redirector_url, str(e))
+        if instance is not None:
+            lgr.warning("Using hard-coded URLs")
+            return instance
+        else:
+            raise RuntimeError(
+                f"Could not retrieve server info from {redirector_url},"
+                " and client does not recognize URL"
+            )
+    server_info = r.json()
+    minversion = Version(server_info["cli-minimal-version"])
+    bad_versions = [Version(v) for v in server_info["cli-bad-versions"]]
```

Comment:
although unlikely, it is possible that server provides some incorrectly formatted semantic version.  That would cause a generic `ValueError` into user's face like
```
$> python3 -c 'from semantic_version import Version; print(Version("0.6.0a1"))'   
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/semantic_version/base.py", line 105, in __init__
    major, minor, patch, prerelease, build = self.parse(version_string, partial)
  File "/usr/lib/python3/dist-packages/semantic_version/base.py", line 303, in parse
    raise ValueError('Invalid version string: %r' % version_string)
ValueError: Invalid version string: '0.6.0a1'
```

I think we better guard for `ValueError` in these two lines, and raise some dedicated or just a ValueError with improved wording, such as 
```python
except ValueError as exc:
    raise ValueError(f"{redirector_url} returned incorrectly formatted version, please contact that server's administrators: {exc}")
```

Reply:
Done.